torch/nn/modules/transformer.py

```diff
@@ -273,10 +278,13 @@ def forward(
             why_not_sparsity_fast_path = f"{str_first_layer} was in training mode"
         elif not src.dim() == 3:
             why_not_sparsity_fast_path = f"input not batched; expected src.dim() of 3 but got {src.dim()}"
-        elif src_key_padding_mask is None:
+        # src_key_padding_mask_ and src_key_padding_mask are the same, minus type
+        # conversion. Chacking <var>_ version for torchscript type refinement
+        # but reporting named input in performance debug hint
+        elif src_key_padding_mask_ is None:
             why_not_sparsity_fast_path = "src_key_padding_mask was None"
         elif (((not hasattr(self, "mask_check")) or self.mask_check)
-                and not torch._nested_tensor_from_mask_left_aligned(src, src_key_padding_mask.logical_not())):
+                and not torch._nested_tensor_from_mask_left_aligned(src, src_key_padding_mask_.logical_not())):
```

Comment:
I'm sorry I didn't catch this in the previous pass but reading this again I am actually somewhat confused here, it seems like we are doing 

1) save `src_key_padding_mask` as `src_key_padding_mask_`
2)`src_key_padding_mask =  _canoncial_mask(...)` which will do bool --> float conversion
3) call `src_key_padding_mask.logical_not()` which will make the mask passed in a boolean mask again

So do we need to save `src_key_padding_mask_` at all? Similarly for `mask`